File: 프로그래머스/lv0/120923. 연속된 수의 합/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        int mid = total/num, cnt=0;
        if(num%2 == 0){
            cnt = num/2;
            answer[cnt] = mid+1;
            answer[cnt-1] = mid;
            for(int i=cnt+1, j=cnt-2; j>=0; i++,j--){
                answer[i] = answer[i-1]+1;
                answer[j] = answer[j+1]-1;
            }
        }else{
            cnt = num/2;
            answer[cnt] = mid;
            for(int i=cnt+1, j=cnt-1; j>=0; i++,j--){
                answer[i] = answer[i-1]+1;
                answer[j] = answer[j+1]-1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120923. 연속된 수의 합/연속된 수의 합.js
function solution(num, total) {
    var answer = [];
    var middle = total/num;
    if(num%2==0){
        for(var i=0; i<num/2; i++){
            answer.unshift(Math.floor(middle)-i);
            answer.push(Math.ceil(middle)+i);
        }
    }else{
        answer.push(middle);
        for(var i=1; i<num/2; i++){
            answer.unshift(middle-i);
            answer.push(middle+i);
        }
    }
    return answer;
}